### PR TITLE
[LFXV2-252] GET committee members

### PR DESCRIPTION
**Pull request by mauriciozanettisalomao**
## Overview

* https://linuxfoundation.atlassian.net/browse/LFXV2-252

This pull request implements and wires up the full "Get Committee Member" use case, replacing placeholder code with real logic and comprehensive tests. The changes ensure that committee member retrieval is validated against committee existence and membership, and that errors are handled correctly throughout the stack.

--- 

# PR: Implement GET /committees/{uid}/members/{member_uid} Endpoint

## Summary

Implemented the GET endpoint for retrieving individual committee members, following the existing solution patterns. The endpoint includes proper validation, error handling, ETag support, and committee existence verification.

### Key Components Implemented:
- **Domain Port**: Updated `CommitteeMemberReader` interface
- **Infrastructure Layer**: NATS KV storage implementation for member retrieval
- **Service Layer**: Committee member orchestrator with validation logic
- **API Handler**: Complete GET endpoint with proper response conversion
- **Comprehensive Tests**: Unit tests for all layers + response conversion methods

### **Test Case 1: Successful Member Retrieval**

**Request:**
```bash
curl -X GET "http://localhost:8080/committees/817b7d4f-7363-4813-9e86-42bb160b3e07/members/88b58d54-aaee-4f01-a3ec-a332ab7f7074?v=1" \
  -H "Authorization: Bearer test-token" \
  -H "Content-Type: application/json"
```

**Response:**
```http
HTTP/1.1 200 OK
Content-Type: application/json
Etag: 4
X-Request-Id: 4f8ecaa0-1ba8-481b-b567-2aec2dc295fc
Content-Length: 566
```

**Response Body:**
```json
{
  "uid": "88b58d54-aaee-4f01-a3ec-a332ab7f7074",
  "username": "govofficial4",
  "email": "gac4@example.com",
  "first_name": "Government",
  "last_name": "Official",
  "job_title": "Director",
  "role": {
    "name": "Director",
    "start_date": "2024-01-01",
    "end_date": ""
  },
  "appointed_by": "Community",
  "status": "Active",
  "voting": {
    "status": "Voting Rep",
    "start_date": "2024-01-01",
    "end_date": ""
  },
  "agency": "GSA",
  "country": "United States",
  "organization": {
    "name": "General Services Administration",
    "website": "https://gsa.gov"
  },
  "created_at": "2025-08-20T16:56:08-03:00",
  "updated_at": "2025-08-20T16:56:08-03:00"
}
```

** Data Integrity Verification:**
NATS storage data matches API response perfectly:
```bash
nats kv get committee-members 88b58d54-aaee-4f01-a3ec-a332ab7f7074 --raw
# Returns identical data structure with proper field mapping
```

---

### **Test Case 2: Committee Not Found (404)**

**Request:**
```bash
curl -X GET "http://localhost:8080/committees/550e8400-e29b-41d4-a716-446655440000/members/88b58d54-aaee-4f01-a3ec-a332ab7f7074?v=1" \
  -H "Authorization: Bearer test-token"
```

**Response:**
```http
HTTP/1.1 404 Not Found
Content-Type: application/json
Goa-Error: NotFound
X-Request-Id: c4047aa0-eee7-4764-8541-54c5f3918c2f
```

**Response Body:**
```json
{
  "message": "committee not found: committee UID: 550e8400-e29b-41d4-a716-446655440000"
}
```

** Validation:** Committee existence check working as designed - returns 404 before attempting member lookup.

---

### **Test Case 3: Member Not Found (404)**

**Request:**
```bash
curl -X GET "http://localhost:8080/committees/817b7d4f-7363-4813-9e86-42bb160b3e07/members/550e8400-e29b-41d4-a716-446655440000?v=1" \
  -H "Authorization: Bearer test-token"
```

**Response:**
```http
HTTP/1.1 404 Not Found
Content-Type: application/json
Goa-Error: NotFound
X-Request-Id: ebedaf36-90d5-4f7f-9519-cfb8f91928ae
```

**Response Body:**
```json
{
  "message": "committee member not found: member UID: 550e8400-e29b-41d4-a716-446655440000"
}
```

** Validation:** Proper member not found error after committee existence is confirmed.

---

### **Test Case 4: Invalid UUID Format (400)**

**Request:**
```bash
curl -X GET "http://localhost:8080/committees/nonexistent-committee/members/88b58d54-aaee-4f01-a3ec-a332ab7f7074?v=1" \
  -H "Authorization: Bearer test-token"
```

**Response:**
```http
HTTP/1.1 400 Bad Request
Content-Type: application/json
X-Request-Id: d1bb3eb8-e824-4724-b8d2-538bd9e5e7a2
```

**Response Body:**
```json
{
  "name": "invalid_format",
  "id": "3H1kBNuL",
  "message": "uid must be formatted as a uuid but got value \"nonexistent-committee\", uuid: nonexistent-committee: invalid UUID length: 21",
  "temporary": false,
  "timeout": false,
  "fault": false
}
```

** Validation:** Proper input validation with clear error messages.

---

### Service Layer Flow:
1. **Committee Validation** ✅ - Verifies committee exists first (returns 404 if not)
2. **Member Retrieval** ✅ - Fetches member from NATS KV store
3. **Ownership Verification** ✅ - Ensures member belongs to requested committee
4. **Response Conversion** ✅ - Maps domain model to API response format
5. **ETag Generation** ✅ - Includes revision for caching support

### Key Features:
- **Security**: Bearer token authentication required
- **Performance**: Direct NATS KV store access with ETag caching
- **Validation**: Strict UUID format validation
- **Logging**: Comprehensive debug and error logging
- **Error Handling**: Proper HTTP status codes and error messages

### Unit Tests Added:
- ✅ **Service Layer**: 6 test cases covering success, errors, and edge cases
- ✅ **Response Conversion**: 11 test cases for payload/domain/response mapping
- ✅ **Infrastructure**: NATS storage implementation with proper error handling

### Integration Tests:
- ✅ **End-to-End API**: All HTTP status codes and response formats verified
- ✅ **Data Integrity**: NATS storage data perfectly matches API responses
- ✅ **Error Scenarios**: Committee not found, member not found, invalid UUIDs

## Endpoint Specification

```
GET /committees/{uid}/members/{member_uid}?v=1
Authorization: Bearer <token>
```

**Success Response (200):**
- Returns complete member information
- Includes ETag header for caching
- Properly formatted timestamps
- All nested objects (role, voting, organization) included

**Error Responses:**
- `400` - Invalid UUID format
- `404` - Committee or member not found
- `401` - Missing/invalid authentication

## ✅ Ready for Production

The implementation is **production-ready** with:
- 🔒 Proper authentication and validation
- 📈 Performance optimized with NATS KV store
- 🛡️ Comprehensive error handling
- 📝 Full test coverage
- 🔄 ETag support for HTTP caching
- 📊 Request tracing with X-Request-Id

---

**All tests passing ✅ | API endpoints verified ✅ | Production ready 🚀**

### CODE REVIEW

File: internal/service/committee_reader_test.go [hunk 1]
```diff
@@ -689,6 +689,228 @@ func TestCommitteeReaderOrchestratorGetBaseAttributeValue(t *testing.T) {
...
+			},
+			committeeUID:  testCommitteeUID,
+			memberUID:     "nonexistent-member-uid",
+			expectedError: true,
+			errorType:     errs.NotFound{},
+			validateMember: func(t *testing.T, member *model.CommitteeMember, revision uint64) {
+				assert.Nil(t, member)
+				assert.Equal(t, uint64(0), revision)
+			},
+		},
```
[the remark this answers is not in the document]
Comment (mauriciozanettisalomao) [reply internal/service/committee_reader_test.go:794]: Thanks!

File: internal/service/committee_reader.go [hunk 2]
```diff
@@ -116,9 +116,52 @@ func (rc *committeeReaderOrchestrator) GetBaseAttributeValue(ctx context.Context
...
+
+	// Verify that the member belongs to the requested committee
+	if committeeMember.CommitteeUID != committeeUID {
+		slog.ErrorContext(ctx, "committee member does not belong to the requested committee",
+			"committee_uid", committeeUID,
+			"member_uid", memberUID,
+			"member_committee_uid", committeeMember.CommitteeUID,
+		)
+		return nil, 0, errors.New("committee member not found in the specified committee")
+	}
```
[the remark this answers is not in the document]
Comment (mauriciozanettisalomao) [reply internal/service/committee_reader.go]: Fixed.

**Review by andrest50 — APPROVED** [review]
